### PR TITLE
Enable ST-Things on esp32-wrover-kit board

### PR DESCRIPTION
**Pull request by zhouxinhe**
- Fix building errors in esp_idf_port module when spiram enabled
- Increase facotry app partion size for large tinyara.elf
- Undefine `HAVE_MEMORY_H` in iotivity_config.h
- Modify wifi manager to notify iotivity the network interface up/down
- Disable wifi scan in soft AP mode in case of esp32 boards
- Add `st_things` configuration for esp32-wrover-kit board

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [650ea74a5b9a5ca9b9ae79b0e952d43c58446462] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [650ea74a5b9a5ca9b9ae79b0e952d43c58446462] - Code Rule Check (C++) OK.

**Comment by zhouxinhe** [comment]
@xixidodo  @Hyunjun85 
Could you please help to review this PR?

File: external/esp_idf_port/esp32/spi_ram/spiram_psram.c [hunk 1]
```diff
@@ -33,16 +33,12 @@
  Driver bits for PSRAM chips (at the moment only the ESP-PSRAM32 chip).
 */
 
-//#include "sdkconfig.h"
 #include "string.h"
 #include "esp_attr.h"
-#include "esp_err.h"
-//#include "esp_types.h"
-//#include "esp_log.h"
+#include <arch/chip/esp_err.h>
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/esp32/spi_ram/spiram_psram.c]: unnecessary here when it's already include in the h file.

File: external/iotivity/iotivity_1.2-rel/resource/c_common/iotivity_config.h [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 #define HAVE_FCNTL_H 1
 
-#define HAVE_MEMORY_H 1
+//#define HAVE_MEMORY_H 1
```
Comment (SRCN-LiQin) [inline external/iotivity/iotivity_1.2-rel/resource/c_common/iotivity_config.h]: should it be commented or set to 0?

File: framework/src/st_things/things_stack/things_stack.c [hunk 3]
```diff
@@ -372,9 +372,11 @@ int things_start_stack(void)
 			if (!things_network_turn_on_soft_ap()) {
 				return 0;
 			}
+#ifndef CONFIG_WIFI_SCAN_IN_SOFT_AP_DISABLED
 			//call wifi scan ap.
 			if (!things_start_scanning_ap())
 				return 0;
```
Comment (SRCN-LiQin) [inline framework/src/st_things/things_stack/things_stack.c]: this if statement have no {}

File: external/iotivity/iotivity_1.2-rel/resource/c_common/iotivity_config.h (hunk 2 above)
Comment (zhouxinhe) [reply external/iotivity/iotivity_1.2-rel/resource/c_common/iotivity_config.h]: It is used as below, we should comment  it.
```c
#ifdef HAVE_MEMORY_H
#include <memory.h>
#endif
```

File: external/esp_idf_port/esp32/spi_ram/spiram_psram.c (hunk 1 above)
Comment (zhouxinhe) [reply external/esp_idf_port/esp32/spi_ram/spiram_psram.c]: I will delete this line, thanks.

File: framework/src/st_things/things_stack/things_stack.c (hunk 3 above)
Comment (zhouxinhe) [reply framework/src/st_things/things_stack/things_stack.c]: I will fix it, thanks

File: external/iotivity/iotivity_1.2-rel/resource/c_common/iotivity_config.h (hunk 2 above)
Comment (xixidodo) [inline external/iotivity/iotivity_1.2-rel/resource/c_common/iotivity_config.h]: will this break ARM build?

Comment (zhouxinhe) [reply external/iotivity/iotivity_1.2-rel/resource/c_common/iotivity_config.h]: There's no problem.
In arm-none-eabi toolchain, `memory.h` only includes `string.h` file.
and HAVE_STRING_H has already been defined in this iotivity_config.h at line 30.

**Comment by seinfra** [comment]
Target : [9310bf895139f9aed29ff7a6260a24caf51516b3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9310bf895139f9aed29ff7a6260a24caf51516b3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f41cd79ef2f6bb511ad70de43cb49130686b0ab5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f41cd79ef2f6bb511ad70de43cb49130686b0ab5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [beb847197413c0f70c48bce939ddcc4a355b3963] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [beb847197413c0f70c48bce939ddcc4a355b3963] - Code Rule Check OK.

**Comment by Hyunjun85** [comment]
In future, Could you please raise PR which was grouped having a relation.
This PR was included commits which had modified serveral layer.

**Comment by zhouxinhe** [comment]
> In future, Could you please raise PR which was grouped having a relation.
> This PR was included commits which had modified serveral layer.

Ok, I will. 
Thanks for your review.

**Pull request merged**